keadm/README.md

```diff
@@ -1,17 +1,258 @@
+
+### Command flags
+The optional flags with this command are shown in below shell
+
+```
+$ sudo kubeedge init --help
+[sudo] password for edgesetup:
+
+kubeedge init command bootstraps KubeEdge's cloud component.
+It checks if the pre-requisites are installed already,
+If not installed, this command will help in download,
+install and execute on the host.
+
+Usage:
+  kubedge init [flags]
+
+Examples:
+
+kubeedge init
+
+
+Flags:
+      --docker-version string[="18.06.0"]          Use this key to download and use the required Docker version (default "18.06.0")
+  -h, --help                                       help for init
+      --kubeedge-version string[="0.3.0-beta.0"]   Use this key to download and use the required KubeEdge version (default "0.3.0-beta.0")
+      --kubernetes-version string[="1.14.1"]       Use this key to download and use the required Kubernetes version (default "1.14.1")
+
+```
+
+1. `--docker-version`, if mentioned with any version > 18.06.0, will install the same on the host. Default is 18.06.0. It is optional.
+2. `--kubernetes-version`, if mentioned with any version > 1.14.1, will install the same on the host. Default is 1.14.1. It is optional.
+                           It will install `kubeadm`, `kubectl` and `kubelet` in this host.
+3. `--kubeedge-version`, if mentioned with any version > 0.2.1, will install the same on the host. Default is 0.3.0-beta.0. It is optional.
+
+command format is
+
+```
+kubedge init --docker-version=<expected version> --kubernetes-version=<expected version> --kubeedge-version=<expected version>
+```
+
+**NOTE:**
+Version mentioned as defaults for Docker and K8S are being tested with.
+
+## Installing KubeEdge Edge component
+
+Referring to `KubeEdge Installer Doc`, the command to install KubeEdge Edge component (edge core) and pre-requisites
+
+- Execute `sudo kubeedge join <flags>`
+
+### Command flags
+
+The optional flags with this command are shown in below shell
+
+```
+$ sudo kubeedge join --help
+[sudo] password for edgesetup: 
+
+"kubeedge join" command bootstraps KubeEdge's edge component.
+It checks if the pre-requisites are installed already,
+If not installed, this command will help in download,
+to install the prerequisites.
+It will help the edge node to connect to the cloud.
+
+
+Usage:
+  kubedge join [flags]
+
+Examples:
+
+kubeedge join --server=<ip:port>
+
+  - For this command --server option is a Mandatory option
+  - This command will download and install the default version of pre-requisites and KubeEdge
+
+kubeedge join --server=10.20.30.40:8080 --docker-version= --kubeedge-version=0.2.1 --kubernetes-version=1.14.1
+kubeedge join --server=10.20.30.40:8080 --docker-version --kubeedge-version=0.2.1 --kubernetes-version=1.14.1
+
+  - In case, any option is used in a format like as shown for "--docker-version" or "--docker-version=", without a value
```

Comment:
why should a user type a flag and leave the flag empty for default value ? 
User should not be mentioning the flag itself to use default

Reply:
just to tell user , above combination is also possible..